### PR TITLE
7339 implement contextual actions inside the commandmenu

### LINKED ISSUE #7339: Implement contextual actions inside the CommandMenu
## Scope and Context
Access the context store to display contextual actions inside the command menu.

Example:
I am on the showPage of a workflow and I open the command menu, I want to have access to all of the actions concerning the workflow:
image

### PR DESCRIPTION
**Pull request by bosiraphael**
Closes #7339

https://github.com/user-attachments/assets/b623caa4-c1b3-448e-8880-4a8301802ba8

### CODE REVIEW

File: packages/twenty-front/src/modules/command-menu/components/CommandMenuActions.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  mainContextStoreComponentInstanceId: string;
+  matchingActionCommands: Command[];
+  setActionCommands: (actionCommands: Command[]) => void;
+}) => {
+  const actionMenuEntries = useRecoilComponentValueV2(
+    actionMenuEntriesComponentSelector,
+    mainContextStoreComponentInstanceId,
+  );
+
+  const actionCommands = useMemo(() => {
```
Comment (Weiko) [inline packages/twenty-front/src/modules/command-menu/components/CommandMenuActions.tsx:23]: useMemo probably not useful here, this part is not hard to compute

File: packages/twenty-front/src/modules/command-menu/components/CommandMenuActions.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      id: actionMenuEntry.key,
+      label: actionMenuEntry.label,
+      Icon: actionMenuEntry.Icon,
+      onCommandClick: actionMenuEntry.onClick,
+      type: CommandType.Action,
+    }));
+  }, [actionMenuEntries]);
+
+  useEffect(() => {
+    setActionCommands(actionCommands);
```
Comment (Weiko) [inline packages/twenty-front/src/modules/command-menu/components/CommandMenuActions.tsx:34]: move logic to useCommandMenu

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/components/DeleteRecordsActionEffect.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { useActionMenuEntries } from '@/action-menu/hooks/useActionMenuEntries';
-import { ActionMenuType } from '@/action-menu/types/ActionMenuType';
+import { ActionMenuContext } from '@/action-menu/states/contexts/ActionMenuContext';
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/action-menu/actions/record-actions/components/DeleteRecordsActionEffect.tsx:2]: move out of states/contexts

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/components/RecordActionMenuEntriesSetter.tsx [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
 import { contextStoreNumberOfSelectedRecordsComponentState } from '@/context-store/states/contextStoreNumberOfSelectedRecordsComponentState';
 import { useObjectMetadataItemById } from '@/object-metadata/hooks/useObjectMetadataItemById';
 import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
 
-export const RecordActionMenuEntriesSetter = ({
-  actionMenuType,
-}: {
-  actionMenuType: ActionMenuType;
-}) => {
+const singleRecordActions = [
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/action-menu/actions/record-actions/components/RecordActionMenuEntriesSetter.tsx:9]: singleRecordActionEffects

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/components/RecordActionMenuEntriesSetter.tsx [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
+  ManageFavoritesActionEffect,
+  ExportRecordsActionEffect,
+  DeleteRecordsActionEffect,
+];
+
+const multipleRecordActions = [
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/action-menu/actions/record-actions/components/RecordActionMenuEntriesSetter.tsx:15]: same

File: packages/twenty-front/src/modules/action-menu/components/RecordIndexActionMenuEffect.tsx [hunk 6]
```diff
@@ -43,5 +48,11 @@ export const RecordIndexActionMenuEffect = () => {
     isDropdownOpen,
   ]);
 
+  useEffect(() => {
+    if (isRightDrawerOpen || isCommandMenuOpened) {
+      closeActionBar();
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/action-menu/components/RecordIndexActionMenuEffect.tsx:53]: move to openCommandMenu() and useOpenRightdrawer()

File: packages/twenty-front/src/modules/action-menu/states/contexts/ActionMenuContext.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { createContext } from 'react';
+
+export type ActionMenuContextType = {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/action-menu/states/contexts/ActionMenuContext.ts:3]: do we need to export it? usually if we export we want a separate file (here in types)

File: packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx [hunk 8]
```diff
@@ -81,13 +85,19 @@ export const RecordIndexPage = () => {
...
-                <SetMainContextStoreComponentInstanceIdEffect />
-                <RecordIndexContainer />
+                <ActionMenuComponentInstanceContext.Provider
+                  value={{
+                    instanceId: `record-index-${objectMetadataItem.id}`,
+                  }}
+                >
+                  <RecordIndexContainerContextStoreObjectMetadataEffect />
+                  <RecordIndexContainerContextStoreNumberOfSelectedRecordsEffect />
+                  <MainContextStoreComponentInstanceIdSetterEffect />
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/object-record/RecordIndexPage.tsx:98]: missing setters in RightDrawer + ShowPage (objectMetadataSetter + main context store)

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**